### PR TITLE
Leave only widget creation tracker kernel pass in flutter frontend server.

### PR DESCRIPTION
**Pull request by aam**
This is follow-up to https://dart-review.googlesource.com/c/sdk/+/43888.

### CODE REVIEW

File: frontend_server/lib/server.dart [hunk 1]
```diff
@@ -5,381 +5,70 @@
...
-    _outputStream ??= stdout;
-    printerFactory ??= new BinaryPrinterFactory();
-    if (trackWidgetCreation) {
-      widgetCreatorTracker = new WidgetCreatorTracker();
-    }
-  }
-
-  StringSink _outputStream;
-  BinaryPrinterFactory printerFactory;
+/// Wrapper around [FrontendCompiler] that adds [wdgetCreatorTracker] kernel
```
Comment (jacob314) [inline frontend_server/lib/server.dart]: nit: typo.
wdgetCreatorTracker instead of widgetCreatorTracker

File: DEPS [hunk 2]
```diff
@@ -31,7 +31,7 @@ vars = {
   # Dart is: https://github.com/dart-lang/sdk/blob/master/DEPS.
   # You can use //tools/dart/create_updated_flutter_deps.py to produce
   # updated revision list of existing dependencies.
-  'dart_revision': '1aaebedd7045a44ba92bb1bb27c3a8bc25d74856',
+  'dart_revision': 'ee15c8eb689791f6777eb3f6b0a1e9c58ff0671b',
```
Comment (alexmarkov) [inline DEPS]: Consider rolling dart-sdk separately (or at least add a note about dart roll into commit message).

File: frontend_server/lib/server.dart [hunk 3]
```diff
@@ -388,15 +77,14 @@ Future<int> starter(
...
-      new _FrontendCompiler(output, printerFactory: binaryPrinterFactory);
-
-      await compiler.compile(
-          Platform.script.toFilePath(), options, generator: generator);
+      options = frontend.argParser.parse(<String>[
+        '--incremental',
+        '--sdk-root=$sdkRoot',
+        '--output-dill=$outputTrainingDill',
+        '--target=flutter']);
+      compiler ??= new _FlutterFrontendCompiler(output, trackWidgetCreation: true);
```
Comment (alexmarkov) [inline frontend_server/lib/server.dart]: Previously, trackWidgetCreation was not enabled in a training run. Is it an intentional change?

File: DEPS (hunk 2 above)
Comment (aam) [reply DEPS]: Yes, my intent is to merge this away once dart rolls.

File: frontend_server/lib/server.dart (hunk 3 above)
Comment (aam) [reply frontend_server/lib/server.dart]: Yes, I thought it make sense to train with trackWidgetCreation enabled.

**Pull request merged**